plugins/hypervisors/xenserver/src/org/apache/cloudstack/storage/motion/XenServerStorageMotionStrategy.java

```diff
@@ -326,7 +326,7 @@ private Answer migrateVmWithVolumesAcrossCluster(VMInstanceVO vm, VirtualMachine
                     volumeToStorageUuid.add(new Pair<>(volumeTo, iqn));
                 }
                 else {
-                    volumeToStorageUuid.add(new Pair<>(volumeTo, ((StoragePool)entry.getValue()).getPath()));
+                    volumeToStorageUuid.add(new Pair<>(volumeTo, ((StoragePool)entry.getValue()).getUuid()));
```

Comment:
This can have compatibility issues wrt non-local disks, can any xenserver guru test/comment - @rafaelweingartner @DaanHoogland etc?

Reply:
I share your concerns, @rhtyd , will run the smoke test and maybe a second run with specific migration tests

Reply:
I do not know what is used in the `path` field there, but if you use the storage UUID, you should not have a problem to find storage repositories in the XenServer host.

Reply:
LGTM, in case of xenserver with storage motion across clusters with local storage the path is set to `lvm`. Error I got was:
```
ERROR [c.c.h.x.r.w.x.XenServer610MigrateWithStorageReceiveCommandWrapper] (DirectAgent-214:ctx-d73466f3) (logid:d54ed2c4) Migration of vm i-2-3-VM with storage failed due to com.cloud.utils.exception.CloudRuntimeException: Can not see storage pool: lvm from on host:fd11220b-db52-42b6-b21d-3f660f7b8a4a
com.cloud.utils.exception.CloudRuntimeException: Can not see storage pool: lvm from on host:fd11220b-db52-42b6-b21d-3f660f7b8a4a
	at com.cloud.hypervisor.xenserver.resource.CitrixResourceBase.getStorageRepository(CitrixResourceBase.java:3249)
	at com.cloud.hypervisor.xenserver.resource.wrapper.xen610.XenServer610MigrateWithStorageReceiveCommandWrapper.execute(XenServer610MigrateWithStorageReceiveCommandWrapper.java:73)
	
```
The regression was introduced in `a30a31c9b7f9b5fe246a02285bf820e7fe9ce16e`. /cc @mike-tutkowski